dspace-api/src/main/java/org/dspace/content/dao/impl/CommunityDAOImpl.java

```diff
@@ -62,19 +60,26 @@ protected CommunityDAOImpl() {
     public List<Community> findAll(Context context, MetadataField sortField, Integer limit, Integer offset)
         throws SQLException {
         StringBuilder queryBuilder = new StringBuilder();
-        queryBuilder.append("SELECT ").append(Community.class.getSimpleName()).append(" FROM Community as ")
-                    .append(Community.class.getSimpleName()).append(" ");
-        addMetadataLeftJoin(queryBuilder, Community.class.getSimpleName(), Arrays.asList(sortField));
-        addMetadataSortQuery(queryBuilder, Arrays.asList(sortField), Collections.EMPTY_LIST);
-
+        queryBuilder.append("SELECT c" +
```

Comment:
Could we add some explanatory comments that describe this query? It's a complex query to follow, and I don't think it will be self explanatory to most developers, so we should be sure to say what it is doing.